### PR TITLE
Mythic Beasts DNS API script

### PR DESCRIPTION
**Pull request by casesolved-co-uk**
Sorry, I can't make sense of the dev CI build errors. Have read the log and the added script works.

### CODE REVIEW

**Comment by Neilpang** [comment]
fix your CI errors first.

**Comment by casesolved-co-uk** [comment]
> fix your CI errors first.

It says "All checks have passed" below?

**Comment by Neilpang** [comment]
https://github.com/casesolved-co-uk/acme.sh/actions

**Comment by Neilpang** [comment]
rebase to the latest  dev branch, and rerun it.

**Comment by casesolved-co-uk** [comment]
Is this good to go now? Solaris still failing, nothing to do with me.

**Comment by Neilpang** [comment]
No,  you must rerun it until it passes or you figure out why.

**Comment by Neilpang** [comment]
don't make empty pushes to run the actions.
Just click the rerun button to run it again.

**Comment by casesolved-co-uk** [comment]
Mythic Beasts have suggested it is failing due to different source IPs being used between obtaining the OAuth2 token and the following DNS requests. Their OAuth2 tokens are tied to the requesting IP.

Normally I would suggest using a session (although no idea if that would solve it) but since these are individual curl requests, I doubt that is an option. Is there anything you can do with the test run environment to affix the IP?

**Comment by casesolved-co-uk** [comment]
@Neilpang   I'm now getting curl error code 60 about invalid SSL certs. Mythic Beasts run a valid Let's Encrypt certificate. Are you sure the Solaris build you're using has up to date Root CA public keys?

**Comment by Neilpang** [comment]
please rebase to latest `dev` branch.  solaris is fixed.

**Comment by casesolved-co-uk** [comment]
@Neilpang   seems to work!

File: dnsapi/dns_mythic_beasts.sh [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  _info "Token OK"
+  return 0
+}
+
+# Github appears to use an outbound proxy for requests which means subsequent requests may not have the same
+# source IP. The standard Mythic Beasts OAuth2 tokens are tied to an IP, meaning github test requests fail
+# authentication. This works arounds this by using an undocumented MB API to obtain a token not tied to an
+# IP just for the github tests.
+_oauth2() {
+  if [ "$TEST_DNS" != "dns_mythic_beasts" ]; then
```
Comment (Neilpang) [inline dnsapi/dns_mythic_beasts.sh:169]: please use "if [ "$GITHUB_ACTIONS" = "true" ]" to check.
https://docs.github.com/en/actions/learn-github-actions/environment-variables

**Comment by casesolved-co-uk** [comment]
LGTM

**Pull request merged**